client.go

```diff
@@ -1297,22 +1324,50 @@ func (l *link) muxReceive(fr performTransfer) error {
 	if err != nil {
 		return err
 	}
-
+	uuid, _ := uuidFromLockTokenBytes(l.msg.DeliveryTag)
+	debug(1, "%s before push to receiver - deliveryCount : %d - linkCredit: %d, len(messages): %d, len(inflight): %d", uuid, l.deliveryCount, l.linkCredit, len(l.messages), len(l.receiver.inFlight.m))
 	// send to receiver, this should never block due to buffering
 	// and flow control.
+	l.addPending(&l.msg)
 	l.messages <- l.msg
 
+	debug(1, "%s after push to receiver - deliveryCount : %d - linkCredit: %d, len(messages): %d, len(inflight): %d", uuid, l.deliveryCount, l.linkCredit, len(l.messages), len(l.receiver.inFlight.m))
+
 	// reset progress
 	l.buf.reset()
 	l.msg = Message{}
 
 	// decrement link-credit after entire message received
 	l.deliveryCount++
 	l.linkCredit--
-
+	debug(1, "%s before exit - deliveryCount : %d - linkCredit: %d, len(messages): %d", uuid, l.deliveryCount, l.linkCredit, len(l.messages))
 	return nil
 }
 
+func uuidFromLockTokenBytes(bytes []byte) (*uuid.UUID, error) {
```

Comment:
brought this in for better logging. might be better in a different file, but none really stood out :)